### DISCUSSION TITLE
For anyone that uses heirline.nvim

### DISCUSSION THREAD

**Discussion by olimorris** [stevearc/overseer.nvim#54] [Ideas]
Just made the switch to `heirline.nvim` and hacked the lualine config:

```lua
local function overseer()
  return {
    condition = function()
      local ok, _ = pcall(require, "overseer")
      if ok then return true end
    end,
    init = function(self)
      self.overseer = require("overseer")
      self.tasks = self.overseer.task_list
      self.STATUS = self.overseer.constants.STATUS
    end,
    static = {
      symbols = {
        ["FAILURE"] = "  ",
        ["CANCELED"] = "  ",
        ["SUCCESS"] = "  ",
        ["RUNNING"] = " 省",
      },
      colors = {
        ["FAILURE"] = "red",
        ["CANCELED"] = "gray",
        ["SUCCESS"] = "green",
        ["RUNNING"] = "yellow",
      },
    },
    {
      condition = function(self) return #self.tasks.list_tasks() > 0 end,
      {
        provider = function(self)
          local tasks_by_status = self.overseer.util.tbl_group_by(self.tasks.list_tasks({ unique = true }), "status")

          for _, status in ipairs(self.STATUS.values) do
            local status_tasks = tasks_by_status[status]
            if self.symbols[status] and status_tasks then
              self.color = self.colors[status]
              return self.symbols[status]
            end
          end
        end,
        hl = function(self) return { fg = self.color } end,
      },
    },
  }
end
```

Gives me this:
![Screen Shot 2022-10-28 at 16 32 00](https://user-images.githubusercontent.com/9512444/198676197-84da5817-a522-492f-8ce6-4dae912dacf8.gif)

**Comment by stevearc** [comment]
Nice! I haven't used heirline myself, but it looks like the focus is on tinkering and writing a lot of it yourself. Does it make sense to put this in the overseer repo somewhere? Is there a canonical path that people would generally look for heirline functions?

**Reply by olimorris** [reply]
Yep that's it. Complete control and absolute versatility. It's been a joy to use. My full config is [here](https://github.com/olimorris/dotfiles/blob/main/.config/nvim/lua/Oli/plugins/statusline.lua). The inheritance model between objects is fantastic.

They have a cookbook so I've asked if they'd accept my code in that.
